src/Number.php

```diff
@@ -168,21 +188,39 @@ public function getIntegerRoundingMultiplier()
      *
      * @return bool
      */
-    private static function validateNumberAsInteger($number)
+    private static function validateAsIntegerPart($number)
     {
-        // Check if number is invalid because of integer overflow
-        $invalid = array_filter(
-            str_split($number, strlen((string) PHP_INT_MAX) - 1),
-            function ($chunk) {
-                // Leading zeros should not invalidate the chunk
-                $chunk = ltrim($chunk, '0');
+        $position = 0;
+        while (isset($number[$position])) {
```

Comment:
Wouldn't this be better with a for/foreach loop, since we use the counter variable (and the value too)?